service/src/test/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiControllerDataprocClusterConnectedTest.java

```diff
@@ -0,0 +1,201 @@
+package bio.terra.workspace.app.controller;
+
+import static bio.terra.workspace.common.utils.MockMvcUtils.assertControlledResourceMetadata;
+import static bio.terra.workspace.common.utils.MockMvcUtils.assertResourceMetadata;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.stairway.FlightDebugInfo;
+import bio.terra.workspace.common.BaseConnectedTest;
+import bio.terra.workspace.common.StairwayTestUtils;
+import bio.terra.workspace.common.utils.MockMvcUtils;
+import bio.terra.workspace.common.utils.TestUtils;
+import bio.terra.workspace.connected.UserAccessUtils;
+import bio.terra.workspace.generated.model.ApiAccessScope;
+import bio.terra.workspace.generated.model.ApiCloudPlatform;
+import bio.terra.workspace.generated.model.ApiErrorReport;
+import bio.terra.workspace.generated.model.ApiGcpDataprocClusterResource;
+import bio.terra.workspace.generated.model.ApiGcpGcsBucketResource;
+import bio.terra.workspace.generated.model.ApiJobReport.StatusEnum;
+import bio.terra.workspace.generated.model.ApiManagedBy;
+import bio.terra.workspace.generated.model.ApiPrivateResourceState;
+import bio.terra.workspace.generated.model.ApiPrivateResourceUser;
+import bio.terra.workspace.generated.model.ApiProperty;
+import bio.terra.workspace.generated.model.ApiResourceLineage;
+import bio.terra.workspace.generated.model.ApiResourceType;
+import bio.terra.workspace.generated.model.ApiStewardshipType;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.workspace.model.WorkspaceConstants;
+import java.util.List;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.web.servlet.MockMvc;
+
+/** Connected tests for controlled Dataproc clusters. */
+// Per-class lifecycle on this test to allow a shared workspace object across tests, which saves
+// time creating and deleting GCP contexts.
+
+@Tag("connectedPlus")
+@TestInstance(Lifecycle.PER_CLASS)
+public class ControlledGcpResourceApiControllerDataprocClusterConnectedTest
+    extends BaseConnectedTest {
+  @Autowired MockMvc mockMvc;
+  @Autowired MockMvcUtils mockMvcUtils;
+  @Autowired UserAccessUtils userAccessUtils;
+  @Autowired JobService jobService;
+
+  private UUID workspaceId;
+  private UUID STAGING_BUCKET_ID;
+  private UUID TEMP_BUCKET_ID;
```

Comment:
nit: these should be `stagingBucketId` and `tempBucketId`, the all-capital naming convention is generally only for constant values